src/main/resources/com/atlassian/jira/cloud/jenkins/configuration/config.jelly

```diff
@@ -1,8 +1,158 @@
+                                                </f:entry>
+                                                <input type="hidden" name="active" value="true" />
+                                                <span id="credentialsId_${loop.index}">${site.credentialsId}</span>
+                                            </div>
+                                            <!-- Readonly view of site data -->
+                                            <tr id="site_${loop.index}" class="site-data-row">
+                                                <td>
+                                                    <span class="siteName" id="siteName_${loop.index}">${site.site}</span>
+                                                </td>
+                                                <td>
+                                                    <span id="webhookUrl_${loop.index}">${site.webhookUrl}</span>
+                                                </td>
+                                                <td class="text-align-right">
+                                                    <a href="#" class="margin-right-1em" onclick="editSite(${loop.index}); return false;">
+                                                        <l:icon src="symbol-edit" class="icon-sm" tooltip="Edit Site details"/>
+                                                    </a>
+                                                    <a href="#" onclick="removeSite(${loop.index}); return false;">
+                                                        <l:icon src="symbol-trash" class="icon-sm" tooltip="Remove Site"/>
+                                                    </a>
+                                                </td>
+                                            </tr>
+                                        </j:forEach>
+                                    </tbody>
+                                </table>
+                            </f:section>
+                        </div>
+
+                        <!-- Add Site Button -->
+                        <input type="button" id="showSiteButton" value="Add Site" class="margin-bottom-2em" onclick="showSiteInputs();" />
+
+                        <div id="siteDataContainer" class="site-data-default" name="sites">
+                            <f:section title="Site Connection details">
+                                <div class="help-area">
+                                    <div class="help display-block">
+                                        <div>A Jira admin will provide the credentials needed to connect to this site.</div>
+                                        <div>Once you've entered these details, select <b>Save</b> to connect.</div>
+                                    </div>
+                                </div>
+
+                                <f:entry title="Site Name:" help="/plugin/atlassian-jira-software-cloud/help-siteName.html" field="site">
```

Comment:
Jenkins does all its validation on the backend, so to take advantage of these ghost functions we need the structure <f:entry... <f:textbox> then it does whimsical magic to link these up to a doCheckXYZ function that we wrote in the management link in PR part 1... its unpleasant but seems to work